### ISSUE TITLE
if one device is down and it comes up at the same time as another goes down in the same polling interval, the first device is marked dead forever until the sq-poller is restarted

### ISSUE THREAD

**Issue by jopietsch** [netenglabs/suzieq#348]
**Describe the bug**

```
vagrant suspend exit02 && sleep 90 && vagrant resume exit02 && vagrant suspend exit01 && sleep 90 && vagrant resume exit01
```

exit02 is down. if I bring it back up in the same polling interval as another device goes down (exit01), then exit02 will stay marked down until I restart the poller. Even if I bring exit02 down and back up again it stays as marked dead.

**Comment by jopietsch** [comment]
```
jpietsch> sqpoller show service=device
    namespace   hostname service  status                  gatherTime  ...                  svcQsize                  wrQsize                nodeQsize pollExcdPeriodCount                        timestamp
0   ospf-ibgp     edge01  device       0    [6.0, 2167.0, 112.60795]  ...           [0.0, 2.0, 0.0]          [0.0, 9.0, 0.5]    [0.0, 8.0, 1.8126507]                   0 2021-03-23 11:32:54.839000-07:00
1   ospf-ibgp     exit01  device       0   [67.0, 27198.0, 416.7333]  ...           [0.0, 5.0, 0.0]  [0.0, 5.0, 0.056640625]   [0.0, 9.0, 0.23046875]                   2 2021-03-23 11:32:59.631000-07:00
2   ospf-ibgp     exit02  device      -1  [67.0, 35812.0, 17946.252]  ...           [0.0, 2.0, 0.0]         [0.0, 21.0, 0.5]         [0.0, 12.0, 5.0]                   1 2021-03-23 11:29:02.050000-07:00
3   ospf-ibgp   internet  device       0    [74.0, 2416.0, 82.72355]  ...           [0.0, 1.0, 0.0]         [0.0, 23.0, 1.0]          [0.0, 8.0, 0.0]                   0 2021-03-23 11:32:57.546000-07:00
4   ospf-ibgp     leaf01  device       0    [72.0, 2632.0, 82.26687]  ...           [0.0, 2.0, 0.0]          [0.0, 7.0, 0.0]         [0.0, 12.0, 0.0]                   0 2021-03-23 11:32:57.476000-07:00
5   ospf-ibgp     leaf02  device       0    [69.0, 2632.0, 80.33557]  ...           [0.0, 1.0, 0.0]          [0.0, 9.0, 0.0]          [0.0, 8.0, 0.0]                   0 2021-03-23 11:32:57.465000-07:00
6   ospf-ibgp     leaf03  device       0    [70.0, 2632.0, 72.80747]  ...           [0.0, 0.0, 0.0]         [0.0, 11.0, 0.0]          [0.0, 8.0, 0.0]                   0 2021-03-23 11:32:57.277000-07:00
7   ospf-ibgp     leaf04  device       0    [72.0, 2631.0, 80.47195]  ...           [0.0, 3.0, 0.0]          [0.0, 5.0, 0.0]          [0.0, 8.0, 0.0]                   0 2021-03-23 11:32:57.735000-07:00
8   ospf-ibgp  server101  device       0    [8.0, 2167.0, 20.289057]  ...   [0.0, 3.0, 0.011820045]          [0.0, 7.0, 0.0]          [0.0, 8.0, 0.0]                   0 2021-03-23 11:32:54.512000-07:00
9   ospf-ibgp  server102  device       0    [6.0, 2166.0, 13.835443]  ...           [0.0, 4.0, 0.0]          [0.0, 4.0, 0.5]   [0.0, 8.0, 0.37890625]                   0 2021-03-23 11:32:54.489000-07:00
10  ospf-ibgp  server103  device       0    [6.0, 2167.0, 14.938388]  ...           [0.0, 4.0, 0.5]          [0.0, 5.0, 0.0]  [0.0, 8.0, 0.029296875]                   0 2021-03-23 11:32:54.488000-07:00
11  ospf-ibgp  server104  device       0    [7.0, 2167.0, 18.917994]  ...  [0.0, 3.0, 0.0033026007]          [0.0, 3.0, 0.0]         [0.0, 12.0, 0.0]                   0 2021-03-23 11:32:54.451000-07:00
12  ospf-ibgp    spine01  device       0    [68.0, 2631.0, 96.31797]  ...           [0.0, 4.0, 0.0]          [0.0, 3.0, 0.0]          [0.0, 9.0, 0.0]                   0 2021-03-23 11:32:57.744000-07:00
13  ospf-ibgp    spine02  device       0    [72.0, 2417.0, 78.93286]  ...           [0.0, 0.0, 0.0]         [0.0, 25.0, 0.0]          [0.0, 8.0, 0.0]                   0 2021-03-23 11:32:57.725000-07:00
```
```
jpietsch> sqpoller show service=device view=all hostname=exit01                                                                                                                                                   Error: 2
   namespace hostname service  status                  gatherTime  ...                 svcQsize                  wrQsize               nodeQsize pollExcdPeriodCount                        timestamp
0  ospf-ibgp   exit01  device       0    [2631.0, 2631.0, 2631.0]  ...          [5.0, 5.0, 5.0]          [1.0, 1.0, 1.0]         [9.0, 9.0, 9.0]                   0 2021-03-23 11:22:54.081000-07:00
1  ospf-ibgp   exit01  device       0   [70.0, 2631.0, 96.547386]  ...          [0.0, 5.0, 0.0]          [0.0, 1.0, 0.0]         [0.0, 9.0, 0.0]                   0 2021-03-23 11:27:55.905000-07:00
2  ospf-ibgp   exit01  device      -1  [70.0, 10009.0, 5041.7246]  ...    [0.0, 5.0, 0.0703125]          [0.0, 1.0, 0.5]         [0.0, 9.0, 0.0]                   0 2021-03-23 11:30:06.446000-07:00
3  ospf-ibgp   exit01  device      -1  [70.0, 25410.0, 15225.862]  ...   [0.0, 5.0, 0.03515625]         [0.0, 1.0, 0.25]         [0.0, 9.0, 3.5]                   1 2021-03-23 11:30:46.860000-07:00
4  ospf-ibgp   exit01  device       1  [70.0, 27198.0, 21211.932]  ...  [0.0, 5.0, 0.017578125]        [0.0, 5.0, 2.625]        [0.0, 9.0, 5.75]                   2 2021-03-23 11:31:29.059000-07:00
5  ospf-ibgp   exit01  device       0   [67.0, 27198.0, 416.7333]  ...          [0.0, 5.0, 0.0]  [0.0, 5.0, 0.056640625]  [0.0, 9.0, 0.23046875]                   2 2021-03-23 11:32:59.631000-07:00

[6 rows x 11 columns]
jpietsch> sqpoller show service=device view=all hostname=exit02
   namespace hostname service  status                  gatherTime                   totalTime         svcQsize             wrQsize           nodeQsize  pollExcdPeriodCount                        timestamp
0  ospf-ibgp   exit02  device       0    [2416.0, 2416.0, 2416.0]    [2416.0, 2416.0, 2416.0]  [2.0, 2.0, 2.0]  [21.0, 21.0, 21.0]  [12.0, 12.0, 12.0]                    0 2021-03-23 11:22:53.866000-07:00
1  ospf-ibgp   exit02  device       0    [67.0, 2416.0, 80.00706]   [67.0, 2416.0, 79.990944]  [0.0, 2.0, 0.0]    [0.0, 21.0, 0.0]    [0.0, 12.0, 0.0]                    0 2021-03-23 11:27:56.156000-07:00
2  ospf-ibgp   exit02  device      -1  [67.0, 35812.0, 17946.252]  [67.0, 35812.0, 17946.248]  [0.0, 2.0, 0.0]    [0.0, 21.0, 0.5]    [0.0, 12.0, 5.0]                    1 2021-03-23 11:29:02.050000-07:00
```

**Comment by ddutt** [comment]
This is fixed with 1db819bfc02df40694ba708a81b74445dce835ce
